### PR TITLE
Dots visualization updated

### PR DESCRIPTION
**Pull request by ryanctan**
Updates from #175.

Includes:
- Legend for colorBy feature
- sizeBy parameters changed: entryLength is now word count such that < 50 words is the minimum size; travelLength is now sum of tour lengths (first month/year subtracted from last month/year) such that < 7 months are minimum size
- groupBy Number of Tours enabled by running new Python script to add numTours variable to the entry data

### CODE REVIEW

File: query.js
Comment (epicfaace) [inline query.js]: Why is this commented out?

Comment (ryanctan) [reply query.js]: the other entryLength uses characters, which is more accurate, the one commented out uses words in case we want to use that instead in the future (depending on if @ceserani wants to use the commented-out thresholds in the visualization sizeBy code)

File: query.js
Comment (epicfaace) [inline query.js]: Why are you calculating this on the fly? I thought this was a field you already added through the python scripts?

Comment (ryanctan) [reply query.js]: Yeah we considered it but decided against it in case the text gets revised in the future — If it’s faster though we can do it on a script

Comment (ceserani) [reply query.js]: i would be in favor to add the word count to the data sheets

Comment (epicfaace) [reply query.js]: It might be faster, but since the "slow" option is not unbearably slow, I think the main consideration is whether we want it to be exported on the data sheets.  @ryanctan

**Comment by ryanctan** [comment]
@epicfaace this should be the final update unless you have anything else

**Comment by epicfaace** [comment]
@ryanctan can you update the PR with a list of what exactly was changed before we merge?

File: python_scripts/numTours.py
Comment (epicfaace) [inline python_scripts/numTours.py]: Can you talk about in the README about the fact that this script is used for the numTours variable used in the visualization?

Comment (epicfaace) [inline python_scripts/numTours.py]: Can you talk about in the README about the fact that this script is used for the numTours variable used in the visualization?

File: query.js
Comment (epicfaace) [inline query.js:529]: If you have time later, it would be great to add an automated test for this function as it's kind of complicated. Not necessary for merging this PR though.

File: query.js
Comment (epicfaace) [inline query.js]:
```suggestion
            return d.tourIndex === i;
```

File: query.js
Comment (epicfaace) [inline query.js]: You can inline this and remove numTravels

```suggestion
            var lastTravel = thisTour[thisTour.length - 1];
```

File: query.js
Comment (epicfaace) [inline query.js]: Can travelStartMonth or travelEndMonth ever be equal to 0? (if so, it may cause bugs here)

File: query.js
Comment (epicfaace) [inline query.js:546]: Did you take into account travelStartYear and travelEndYear?

File: query.js
Comment (epicfaace) [inline query.js:560]: Why did you conclude that # of words is better than # of characters for "entry length"?

File: query.js
Comment (epicfaace) [inline query.js:532]: Why did you remove this code which takes into account travelEndYear and travelStartYear?

File: python_scripts/numTours.py
Comment (epicfaace) [inline python_scripts/numTours.py]: Remove this line

File: python_scripts/numTours.py
Comment (epicfaace) [inline python_scripts/numTours.py]: Remove this line

**Comment by ryanctan** [comment]
@epicfaace done!

File: public/visualization.component.ts
Comment (epicfaace) [inline public/visualization.component.ts]: I think it might be better to rename these to `COLOR_MALE`, etc. That way autocomplete would be useful when you type in "COLOR_"

File: public/visualization.component.ts
Comment (epicfaace) [inline public/visualization.component.ts]: Can we remove this note about char count -> word count?

File: public/visualization.component.ts
Comment (epicfaace) [inline public/visualization.component.ts:338]: You can remove this line

**Comment by epicfaace** [comment]
Feel free to merge this if you want to deploy it to alpha and address my comments later.

File: query.js
Comment (ryanctan) [reply query.js]: Yes, but the date declaration/arithmetic handles it automatically

File: query.js
Comment (ryanctan) [reply query.js:546]: This was changed to tourStartFrom and tourEndTo because of a data error that said Edward Bird (422)'s last travel was in 1794 but last tour date was 1734, making his dot noticeably larger

File: query.js
Comment (ryanctan) [reply query.js:560]: Per @ceserani's preference

File: query.js
Comment (ryanctan) [reply query.js:532]: Per the comment in this PR on query.js 540-543, since we only care about the years in relation to the tours, I omitted it

**Comment by ryanctan** [comment]
@epicfaace those changes have been made, though I think I still need an approving review before we can merge

File: query.js
Comment (epicfaace) [reply query.js:546]: Got it, thanks

**Review by epicfaace — APPROVED** [review]

File: query.js
Comment (epicfaace) [reply query.js]: What is supposed to happen when travelStartMonth or travelEndMonth are equal to 0?

Right now, if either is the case, then the if statement will evaluate to false and the difference won't be added to `accum`.

Comment (ryanctan) [reply query.js]: Ohh, I gotcha. Just committed a bug fix

**Pull request merged**
